core/src/main/java/org/kohsuke/stapler/HttpResponses.java

```diff
@@ -88,16 +96,26 @@ public static HttpResponseException error(Throwable cause) {
     }
 
     public static HttpResponseException error(final int code, final Throwable cause) {
-        return new HttpResponseException(cause) {
-            public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node) throws IOException, ServletException {
-                rsp.setStatus(code);
-
-                rsp.setContentType("text/plain;charset=UTF-8");
-                PrintWriter w = new PrintWriter(rsp.getWriter());
-                cause.printStackTrace(w);
-                w.close();
+        HttpResponseException responseException = null;
+        for (ErrorCustomizer handler : ServiceLoader.load(ErrorCustomizer.class)) {
+            responseException = handler.handleError(code, cause);
+            if (responseException != null) {
+                break;
```

Comment:
❌ Any reason to not follow the existing pattern from Daniel? I think especially about the "first to handle" and the classload parameter in `.load()`.
If there are problems the other code should be changed and both using the same approach. So that we can prevent discrepancies.

`break` => For me the break is a no-no but actually inside Stapler, there are multiple occurrences, so it's "fine". Here you do not have a real interest to use `break` instead of just `return` except in cases where you have multiple ErrorCustomizer and wants to use the last one instead of the first one to returns a non-null value.

Reply:
This method doesn't have request, which is used in `RequirePOST` to obtain the class loader. The `HttpResponseException.generateResponse()` method does have the request but that doesn't help us for obtaining the handler.

Reply:
There is a coding style that says a method should only ever return from one place. Do you require a different style?

Reply:
I don't understand what you want here about "first to handle" and similar patterns. Do you want this to throw an exception, as in `RequirePOST`, instead of returning one?

Reply:
The request could be retrieved using static call to Stapler, but if it's not required, no need to change.

I am more concerned by the "first to handle" -> in RequirePOST the first error customizer will provode the result while in your proposal, it's the last one.

Of course it works but will cause more headaches if something different happen depending on which customizer you are using.

Let's move forward.

Reply:
After quick live discussion, I misinterpret the break as a continue and my bad, the behavior was just as the one from Daniel. So good to go.